grayskull/pypi/pypi.py

```diff
@@ -59,21 +58,49 @@ def __init__(
         self["build"]["script"] = "<{ PYTHON }} -m pip install . -vv"
 
     @staticmethod
-    def _download_sdist_pkg(sdist_url: str, dest: str):
+    def _get_latest_version_of_github_repo(git_url: str) -> str:
+        """get the latest version of the github repository using github api
+        """
+        latest_version_api_url = git_url.replace("www.github.com", "api.github.com/repos", 1) + "/releases/latest"
+        response = requests.get(latest_version_api_url)
```

Comment:
Always check for the http return code, because requests won't fail if the request failed (e.g. 404 lets the code proceed, but it failed).

```suggestion
        response = requests.get(latest_version_api_url)
        response.raise_for_status()
```